src/adagrams.js

```diff
@@ -1,15 +1,158 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+class Adagrams {
+  drawLetters = () => {
+    // Implement this method for wave 1
+    const poolOfLettersDict = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
+    const lettersInHand = new Array();
+    let counter = 0;
+    const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    while (counter < 10) {
+      let choice = characters.charAt(Math.floor(Math.random() * 26));
+      if (poolOfLettersDict[choice] > 0) {
+        lettersInHand.push(choice);
+        counter += 1;
+        poolOfLettersDict[choice] -= 1;
+      }
+    }
+    return lettersInHand;
+  };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  usesAvailableLetters = (input, lettersInHand) => {
+    const dictLettersInHand = {};
+    for (let i = 0; i < lettersInHand.length; i++) {
+      if (lettersInHand[i] in dictLettersInHand) {
+        let val = dictLettersInHand[lettersInHand[i]];
+        dictLettersInHand[lettersInHand[i]] = val + 1;
+      } else {
+        dictLettersInHand[lettersInHand[i]] = 1;
+      }
+    }
+    for (let i = 0; i < input.length; i++) {
+      if (input[i] in dictLettersInHand) {
+        let val1 = dictLettersInHand[input[i]];
+        val1 = val1 - 1;
+        dictLettersInHand[input[i]] = val1;
+        if (val1 === 0) {
+          delete dictLettersInHand[input[i]];
+        }
+      } else {
+        return false;
+      }
+    }
+    return true;
+  };
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+  scoreWord = (word) => {
+    // Implement this method for wave 3
+    let scoreDict = {
```

Comment:
This can be declared outside of the function so that it doesn't have to be recreated each time the function is called.